src/Sulu/Bundle/CustomUrlBundle/Controller/CustomUrlController.php

```diff
@@ -0,0 +1,149 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CustomUrlBundle\Controller;
+
+use FOS\RestBundle\Controller\Annotations\RouteResource;
+use Hateoas\Representation\CollectionRepresentation;
+use Hateoas\Representation\RouteAwareRepresentation;
+use Sulu\Component\Rest\RequestParametersTrait;
+use Sulu\Component\Rest\RestController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Provides rest api for custom-urls.
+ *
+ * @RouteResource("custom-urls")
+ */
+class CustomUrlController extends RestController
+{
+    use RequestParametersTrait;
+
+    private static $relationName = 'custom-urls';
+
+    /**
+     * Returns a list of custom-urls.
+     *
+     * @param string $webspaceKey
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function cgetAction($webspaceKey, Request $request)
+    {
+        // TODO pagination
+
+        $result = $this->get('sulu_custom_urls.manager')->readList($webspaceKey);
+
+        $list = new RouteAwareRepresentation(
+            new CollectionRepresentation($result, self::$relationName),
+            'cget_webspace_custom-urls',
+            array_merge($request->request->all(), ['webspaceKey' => $webspaceKey])
+        );
+
+        return $this->handleView($this->view($list));
+    }
+
+    /**
+     * Returns a single custom-url identified by uuid.
+     *
+     * @param string $webspaceKey
+     * @param string $uuid
+     *
+     * @return Response
+     */
+    public function getAction($webspaceKey, $uuid, Request $request)
+    {
+        $document = $this->get('sulu_custom_urls.manager')->read($uuid, $this->getLocale($request));
+        // FIXME without this target-document will not be loaded (for serialization)
+        $document->getTarget()->getTitle();
```

Comment:
@dantleech @danrot do you have any idea why proxy object will no be hydrated during the serialization process?